### PR TITLE
Implementation of LDAP filters

### PR DESCRIPTION
**Pull request by tcalmant**
* Added filtered snapshot support to core
* Added filters.core to support multiple filter languages
* Added implementation of LDAP filters

Not supported yet:
* GeoFiltering (complex in LDAP syntax)
* Escaping characters in LDAP value

### CODE REVIEW

File: prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/impl/snapshot/package-info.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+* terms of the Eclipse Public License 2.0 which is available at
+* https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors: Kentyou - initial implementation
+**********************************************************************/
+@org.osgi.annotation.bundle.Export
+@org.osgi.annotation.versioning.Version("0.2.0")
+package org.eclipse.sensinact.prototype.impl.snapshot;
```
Comment (timothyjward) [inline prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/impl/snapshot/package-info.java:14]: We should definitely not be exporting the implementation package

File: prototype/core/api/src/main/java/org/eclipse/sensinact/prototype/snapshot/package-info.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*********************************************************************
+* Copyright (c) 2022 Contributors to the Eclipse Foundation.
```
Comment (timothyjward) [inline prototype/core/api/src/main/java/org/eclipse/sensinact/prototype/snapshot/package-info.java:2]:
```suggestion
* Copyright (c) 2023 Contributors to the Eclipse Foundation.
```

File: prototype/core/impl/src/test/java/org/eclipse/sensinact/prototype/integration/filter/FilterTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/*********************************************************************
+* Copyright (c) 2022 Contributors to the Eclipse Foundation.
```
Comment (timothyjward) [inline prototype/core/impl/src/test/java/org/eclipse/sensinact/prototype/integration/filter/FilterTest.java:2]:
```suggestion
* Copyright (c) 2023 Contributors to the Eclipse Foundation.
```

File: prototype/northbound/filters/filters.core/src/main/java/org/eclipse/sensinact/northbound/filters/api/FilterCommandHelper.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+/*********************************************************************
+* Copyright (c) 2022 Contributors to the Eclipse Foundation.
```
Comment (timothyjward) [inline prototype/northbound/filters/filters.core/src/main/java/org/eclipse/sensinact/northbound/filters/api/FilterCommandHelper.java:2]:
```suggestion
* Copyright (c) 2023 Contributors to the Eclipse Foundation.
```

File: prototype/northbound/filters/filters.core/src/main/java/org/eclipse/sensinact/northbound/filters/api/FilterException.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+/*********************************************************************
+* Copyright (c) 2022 Contributors to the Eclipse Foundation.
```
Comment (timothyjward) [inline prototype/northbound/filters/filters.core/src/main/java/org/eclipse/sensinact/northbound/filters/api/FilterException.java:2]:
```suggestion
* Copyright (c) 2023 Contributors to the Eclipse Foundation.
```

File: prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/impl/snapshot/package-info.java (hunk 1 above)
Comment (tcalmant) [reply prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/impl/snapshot/package-info.java:14]: Definitely a debugging relic

**Review by timothyjward — APPROVED** [review]

**Pull request merged**